tmt/steps/provision/mrack.py

```diff
@@ -357,6 +358,42 @@ def _transform_virtualization_is_virtualized(
     return _transform_unsupported(constraint, logger)
 
 
+def _transform_zcrypt_adapter(
+        constraint: tmt.hardware.TextConstraint,
+        logger: tmt.log.Logger) -> MrackBaseHWElement:
+    beaker_operator, actual_value, negate = operator_to_beaker_op(
+        constraint.operator,
+        constraint.value)
+
+    if negate:
+        return MrackHWNotGroup(children=[
+            MrackHWGroup(
+                'system',
+                children=[MrackHWKeyValue('ZCRYPT_MODEL', beaker_operator, actual_value)])])
+
+    return MrackHWGroup(
+        'system',
+        children=[MrackHWKeyValue('ZCRYPT_MODEL', beaker_operator, actual_value)])
+
+
+def _transform_zcrypt_mode(
+        constraint: tmt.hardware.FlagConstraint,
```

Comment:
```suggestion
        constraint: tmt.hardware.TextConstraint,
```

The `mode` key is not a boolean flag, is it?

Reply:
Fixed & rebased.